Makefile.defs

```diff
@@ -78,11 +78,9 @@ GO_BUILD_LDFLAGS += -X "github.com/cilium/cilium/pkg/envoy.RequiredEnvoyVersionS
 
 # Use git only if in a Git repo, otherwise depend on file BPF_SRCFILES existing
 ifneq ($(wildcard $(dir $(lastword $(MAKEFILE_LIST)))/.git),)
-	BPF_FILES_EVAL = $(shell git ls-files $(ROOT_DIR)/bpf/ | grep -v .gitignore | tr "\n" ' ')
-	BPF_FILES ?= $(BPF_FILES_EVAL)
-	BPF_SRCFILES := $(subst ../,,$(BPF_FILES))
+	BPF_SRCFILES := $(shell git ls-files $(ROOT_DIR)/bpf/ | grep -v .gitignore | tr "\n" ' ')
 else
-	BPF_SRCFILES = $(shell cat $(ROOT_DIR)/BPF_SRCFILES)
+	BPF_SRCFILES := $(shell cd $(ROOT_DIR) && find bpf/ -type f | grep -v .gitignore | tr "\n" ' ')
```

Comment:
```suggestion
	BPF_SRCFILES := $(shell find $(ROOT_DIR)/bpf/ -type f | grep -v .gitignore | tr "\n" ' ')
```
Note: untested but it should work. I think it's better to avoid using `cd ...`.

Reply:
The point is that we want to search`${ROOT_DIR}/bpf`, but `find ${ROOT_DIR}/bpf` yields having `${ROOT_DIR}` prefix in all of the results. And in non-git case inside a container `ROOT_DIR` was not set to `../`, it was actually absolute path, namely `/go/src/github.com/cilium/cilium`.

It does look the whole thing is more convoluted then I expected it to be, I might have to rethink it or even leave for later.

Reply:
I've reverted back to `subst`, but now the idea is that `ROOT_DIR` is being stripped-off explicitly, not `../`.